sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -0,0 +1,391 @@
+@pytest.mark.parametrize('solver', GLM_SOLVERS)
+def test_normal_ridge_comparison(n_samples, n_features, fit_intercept, solver):
+    """Test ridge regression for Normal distributions.
+
+    Case n_samples >> n_features
+
+    Compare to test_ridge in test_ridge.py.
+    """
+    alpha = 1.0
+    n_predict = 10
+    X, y, coef = make_regression(n_samples=n_samples+n_predict,
+                                 n_features=n_features,
+                                 n_informative=n_features-2, noise=0.5,
+                                 coef=True, random_state=42)
+    y = y[0:n_samples]
+    X, T = X[0:n_samples], X[n_samples:]
+
+    if n_samples > n_features:
+        ridge_params = {"solver": "svd"}
+    else:
+        ridge_params = {"solver": "sag", "max_iter": 10000, "tol": 1e-9}
+
+    # GLM has 1/(2*n) * Loss + 1/2*L2, Ridge has Loss + L2
+    ridge = Ridge(alpha=alpha*n_samples, normalize=False,
+                  random_state=42, **ridge_params)
+    ridge.fit(X, y)
+
+    glm = GeneralizedLinearRegressor(alpha=1.0, family='normal',
+                                     link='identity', fit_intercept=True,
+                                     max_iter=300, solver=solver, tol=1e-6,
+                                     check_input=False)
+    glm.fit(X, y)
+    assert glm.coef_.shape == (X.shape[1], )
+    assert_allclose(glm.coef_, ridge.coef_, rtol=5e-6)
+    assert_allclose(glm.intercept_, ridge.intercept_, rtol=1e-5)
+    assert_allclose(glm.predict(T), ridge.predict(T), rtol=1e-5)
+
+
+@pytest.mark.parametrize('solver, tol', [('lbfgs', 1e-7)])
+def test_poisson_ridge(solver, tol):
+    """Test ridge regression with poisson family and LogLink.
+
+    Compare to R's glmnet"""
+    # library("glmnet")
+    # options(digits=10)
+    # df <- data.frame(a=c(-2,-1,1,2), b=c(0,0,1,1), y=c(0,1,1,2))
+    # x <- data.matrix(df[,c("a", "b")])
+    # y <- df$y
+    # fit <- glmnet(x=x, y=y, alpha=0, intercept=T, family="poisson",
+    #               standardize=F, thresh=1e-10, nlambda=10000)
+    # coef(fit, s=1)
+    # (Intercept) -0.12889386979
+    # a            0.29019207995
+    # b            0.03741173122
+    X = np.array([[-2, -1, 1, 2], [0, 0, 1, 1]]).T
+    y = np.array([0, 1, 1, 2])
+    glm = GeneralizedLinearRegressor(alpha=1,
+                                     fit_intercept=True, family='poisson',
+                                     link='log', tol=1e-7,
+                                     solver=solver, max_iter=300,
+                                     )
+    glm.fit(X, y)
+    assert_allclose(glm.intercept_, -0.12889386979, rtol=1e-5)
+    assert_allclose(glm.coef_, [0.29019207995, 0.03741173122], rtol=1e-5)
+
+
+@pytest.mark.parametrize(
+        "params",
+        [
+            {"solver": "lbfgs"},
+        ],
+        ids=lambda params: ', '.join("%s=%s" % (key, val)
+                                     for key,  val in params.items())
+)
+def test_solver_equivalence(params, regression_data):
```

Comment:
I'd just remove this test. It might be useful for later, but there's a high chance we'll just forget it's here and write a new one anyway.